src/kem/sike/P434/AMD64/fp_x64_asm.S

```diff
@@ -0,0 +1,939 @@
+//  NOTE: a=c or b=c are not allowed
+//***********************************************************************
+.global mul434_asm
+mul434_asm:
+
+  ret
+
+# error "CONFIGURATION NOT SUPPORTED. TRY USE_MULX=TRUE"
+
+#endif
+
+
+#ifdef _MULX_
+
+///////////////////////////////////////////////////////////////// MACRO
+// Schoolbook integer multiplication
+// Inputs:  memory pointers M0 and M1
+// Outputs: regs T0:T5
+// Temps:   regs T7:T6
+/////////////////////////////////////////////////////////////////
+.macro MUL64x256_SCHOOL M0, M1, T0, T1, T2, T3, T4, T5 
+    mov    rdx, \M0
+    mulx   \T1, \T0, \M1       // T0 <- C0_final    
+    mulx   \T2, \T4, 8\M1
+    xor    rax, rax
+    mulx   \T3, \T5, 16\M1 
+    ADD1   \T1, \T4            // T1 <- C1_final   
+    ADC1   \T2, \T5            // T2 <- C2_final 
+    mulx   \T4, \T5, 24\M1
+    ADC1   \T3, \T5            // T3 <- C3_final
+    ADC1   \T4, rax            // T4 <- C4_final
+.endm
+
+#ifdef _ADX_
+.macro MUL128x256_SCHOOL M0, M1, T0, T1, T2, T3, T4, T5, T6 
+    mov    rdx, \M0
+    mulx   \T1, \T0, \M1       // T0 <- C0_final    
+    mulx   \T2, \T4, 8\M1
+    xor    rax, rax
+    mulx   \T3, \T5, 16\M1 
+    ADD1   \T1, \T4               
+    ADC1   \T2, \T5     
+    mulx   \T4, \T5, 24\M1
+    ADC1   \T3, \T5 
+    ADC1   \T4, rax   
+    
+    xor    rax, rax
+    mov    rdx, 8\M0 
+    mulx   \T6, \T5, \M1 
+    ADD2   \T1, \T5            // T1 <- C1_final 
+    ADC2   \T2, \T6     
+    mulx   \T5, \T6, 8\M1
+    ADC2   \T3, \T5 
+    ADD1   \T2, \T6        
+    mulx   \T5, \T6, 16\M1
+    ADC2   \T4, \T5 
+    ADC1   \T3, \T6     
+    mulx   \T5, \T6, 24\M1   
+    ADC2   \T5, rax         
+    ADC1   \T4, \T6  
+    ADC1   \T5, rax 
+.endm
+
+#else
+
+.macro MUL128x256_SCHOOL M0, M1, T0, T1, T2, T3, T4, T5, T6 
+    mov    rdx, \M0
+    mulx   \T1, \T0, \M1       // T0 <- C0_final    
+    mulx   \T2, \T4, 8\M1
+    mulx   \T3, \T5, 16\M1 
+    add    \T1, \T4               
+    adc    \T2, \T5     
+    mulx   \T4, \T5, 24\M1
+    adc    \T3, \T5 
+    adc    \T4, 0   
+    
+    mov    rdx, 8\M0 
+    mulx   \T6, \T5, \M1 
+    add    \T1, \T5            // T1 <- C1_final 
+    adc    \T2, \T6     
+    mulx   \T5, \T6, 8\M1
+    adc    \T3, \T5       
+    mulx   \T5, rax, 16\M1
+    adc    \T4, \T5     
+    mulx   \T5, rdx, 24\M1 
+    adc    \T5, 0
+    add    \T2, \T6  
+    adc    \T3, rax        
+    adc    \T4, rdx  
+    adc    \T5, 0 
+.endm
+#endif
+
+  
+//**************************************************************************************
+//  Montgomery reduction
+//  Based on method described in Faz-Hernandez et al. https://eprint.iacr.org/2017/1015
+//  Operation: c [reg_p2] = a [reg_p1]
+//  NOTE: a=c is not allowed
+//************************************************************************************** 
+.global rdc434_asm
+rdc434_asm:
+    push   r12
+    push   r13 
+
+    // a[0-1] x p434p1_nz --> result: r8:r13 
```

Comment:
I think p434 doesn't use ``p434p1_nz``. it uses just ``p434+1``

Reply:
This code is coming from https://github.com/Microsoft/PQCrypto-SIDH so I'd want to check with Patrick Longa to see what should be done here.

Reply:
ok, I'll open PR there

Reply:
Thanks Kris, I'll follow up with Patrick to see how this can be integrated. As mentioned by Douglas, we hope to keep the OQS version as un-modified as possible from the original SIKE library. I'm mostly on vacation this week, so I hope to get to this next week.

Reply:
no worries, it's just a comment... I was wondering if I should even bother to open PR.

Reply:
I've pointed your comments to Patrick.